packages/main/cypress/specs/Input.cy.ts

```diff
@@ -128,4 +128,41 @@ describe("Input Tests", () => {
 			.find("ul")
 			.should("not.have.attr", "tabindex", "0");
 	});
+
+	it("tests submit and change event order", () => {
+		cy.mount(html`
+			<form>
+				<ui5-input />
+			</form>
+		`);
+
+		cy.get("form")
+			.as("form");
+
+		cy.get("[ui5-input]")
+			.as("input");
+
+		// spy change event
+		cy.get<Input>("@input")
+			.then($input => {
+				$input.get(0).addEventListener("change", cy.spy().as("change"));
+			});
+
+		// spy submit event and prevent it
+		cy.get("@form")
+			.then($form => {
+				$form.get(0).addEventListener("submit", e => e.preventDefault());
+				$form.get(0).addEventListener("submit", cy.spy().as("submit"));
+			});
+
+		// check if submit is triggered after change
+		cy.get<Input>("@input")
+			.shadow()
+			.find("input")
+			.type("test{enter}{enter}");
```

Comment:
I think the test setup is not quite right. Since there are two ways in the current implementation which trigger submit, it's unclear which way is triggering what.

So 
* the first test setup should be `test{enter}` only
* then the `have.been.called` tests
* then again with another `{enter}`. 
* and `have.been.called` again...

Reply:
Might be improved, sure